orca/requester.py

```diff
@@ -13,33 +14,124 @@
 
 def file_from_opendap(url, outdir, outfile):
     """Write to file from OPeNDAP link"""
-    urls = bisect_request(url)
+    file_extension = url.split(".")[-1].split("?")[0]
+    if file_extension != "nc":
+        outpath = to_file(url, outdir, outfile, nc=False)
+    else:
+        urls = bisect_request(url)
 
-    logger.debug(f"Downloading from {len(urls)} URL(s): {urls}")
-    dataset = open_mfdataset(urls, combine="nested", concat_dim="time")
-    outpath = to_file(dataset, outdir, outfile)
+        logger.debug(f"Downloading from {len(urls)} URL(s): {urls}")
+        if len(urls) == 1:
```

Comment:
Is this necessary? Does `open_mfdataset()` not handle the case where `urls` is of length 1? The other consideration here, is that if `urls` is of length 1, then `orca` has no raison d'être. `orca` should simply issue a redirect (say an HTTP 301?) directly to THREDDS.

Reply:
I have it this way because if there's only one url, it'll add an extra time dimension if it doesn't exist:
```python
>>> from xarray import open_mfdataset
>>> url = "https://docker-dev03.pcic.uvic.ca/twitcher/ows/proxy/thredds/dodsC/datasets/storage/data/projects/hydrology/vic_gen2/input/routing/columbia/parameters/domain.pnw.pcic.20170927.nc"
>>> dataset = open_mfdataset([url], combine="nested", concat_dim="time")
>>> dataset.dims
Frozen({'lon': 245, 'lat': 195, 'time': 1})
```
As for the redirect, `open_dataset` will get the data from THREDDS, and then I'm saving it to a file in local storage.